nifi-nar-bundles/nifi-websocket-bundle/nifi-websocket-services-jetty/src/main/java/org/apache/nifi/websocket/jetty/JettyWebSocketClient.java

```diff
@@ -248,6 +268,44 @@ public void startClient(final ConfigurationContext context) throws Exception {
         }, sessionMaintenanceInterval, sessionMaintenanceInterval, TimeUnit.MILLISECONDS);
     }
 
+    private URI getEndpoint(final ConfigurationContext context) throws URISyntaxException {
+        final StringBuilder result = new StringBuilder(context.getProperty(WS_URI).evaluateAttributeExpressions(new HashMap<>()).getValue());
+        final List<PropertyDescriptor> dynamicProperties = context.getProperties()
+                .keySet()
+                .stream()
+                .filter(PropertyDescriptor::isDynamic)
+                .collect(Collectors.toList());
+
+        final Map<String, String> queryParameters = new HashMap<>();
+        dynamicProperties.forEach((descriptor) -> {
+            final PropertyValue propertyValue = context.getProperty(descriptor);
+            if (descriptor.isSensitive()) {
+                final String propertyName = org.apache.commons.lang3.StringUtils.substringAfter(descriptor.getName(), SENSITIVE_PROPERTY_PREFIX);
+                queryParameters.put(propertyName, propertyValue.getValue());
+            } else {
+                queryParameters.put(descriptor.getName(), propertyValue.evaluateAttributeExpressions().getValue());
+            }
+        });
+
+        if (!queryParameters.isEmpty()) {
+            final List<String> parameters = new LinkedList<>();
+
+            try {
+                for (final Map.Entry<String, String> parameter : queryParameters.entrySet()) {
+                    parameters.add(URLEncoder.encode(parameter.getKey(), "UTF-8") + '=' + URLEncoder.encode(parameter.getValue(), "UTF-8"));
```

Comment:
For clarity, it would be helpful to declare the key and value elements separately.  The `UTF-8` string can be replaced with `StandardCharsets.UTF_8`.